runners/katacoda/index.ts

```diff
@@ -416,4 +375,25 @@ export class Katacoda extends Runner {
         return {terminalId: this.terminalCounter, isRunning: false};
     }
 
+    private getStepsCount(runCommand: RunCommand): number {
+        let returnCount = runCommand.stepIndex == this.stepsCount - 1 ? this.stepsCount : ++this.stepsCount;
```

Comment:
Because you use the prefix increment operator now to increment the stepCount, the false time is written into the index.json. Just remove the '-1' part of the second argument in the KatacodaTools.generateIndexJson call in the destroy method.